packages/renderer/src/lib/dialogs/TextHighLight.svelte

```diff
@@ -0,0 +1,39 @@
+<script lang="ts">
+interface Props {
+  text: string;
+  query: string;
+}
+
+let { text, query }: Props = $props();
+
+let highlightedParts = $derived(highlightText(text, query));
```

Comment:
nitpick: my preference is to use a `$derived.by` here without the intermediate function